### ISSUE TITLE
Tests appear to use the production database

### ISSUE THREAD

**Issue by allanderek** [karlorg/drunken-octo-avenger#45]
The tests appear to be using the production database.
I intend to look at this when I fix the tests for coverage analysis.

**Comment by karlorg** [comment]
The casper tests, you mean?  It's hard to see how we'd avoid it since we're trying to simulate an ordinary user (aside from some shortcuts to, eg., create a login session without having to go through the whole login process every time; and now that Persona is gone those might not even be necessary now).  Would you send a /test_fake_db request or something to indicate that this session should use a testing db?  Seems complicated to track.

The actions taken by the casper tests shouldn't harm the database anyway; aside from the aforementioned shortcuts they interact with the db only through ordinary actions a normal user could make (unlike the Python tests, which access the db directly in many cases).

**Comment by allanderek** [comment]
Well it seems wrong to have the tests use the production database. Here is an example scenario. I wish to run the tests on the server before I actually deploy some new code. So I run the tests. I've added a feature where by you can get a list of current players open to new games. I have a test for this feature. So our test creates a new player and that player puts themselves on the open for a game list. They then of course complete the rest of the tests and delete the user, but in the meantime and *actual* user has started a game with them. Okay so this doesn't destroy anything but it is not desired behaviour.

You could think of many other scenarios, such a test for starting a new tournament, or something that is just generally visible.

However, on the 'improve-tests' branch I have fixed it so that all the tests use a temporary database anyway, so once (if) we merge that branch then we can close this issue.

**Comment by karlorg** [comment]
Oh do you mean you're running the tests from the command line on the remote server?  I thought you meant by pointing Casper, running on your local machine, to the live site for the most realistic tests possible.  Admittedly that doesn't work right now because the casper tests use those test-only routes :/  (Maybe we could create a subset of casper tests that use only regular routes for this purpose.)

Yeah if you're confident running the test server on the remote site won't interfere with the production server, I don't see why this change would be a problem.

**Comment by allanderek** [comment]
Well I have not been running the tests on the remote/production server precisely because until now the tests have used the production database. But that is something we *should* do. In the past I made an error because I re-launched the web-app and had forgotten to do a 'python manage.py coffeebuild'. Now, yes, the deploy script should take care of this for us, but you cannot always be sure you have fixed the deploy script properly. So the deploy script really should run the tests locally (that is in place on the production server), to check everything is fine.

I'm sure we're not doing something that is common, like using Docker, which sort of solves this problem for you.

But as to your suggestion to create a sub-set of tests that can actually access the production server, (even if from your local machine), YES, that sounds like a great idea. Ultimately it would be great to set-up a cron job that did that every day and sent emails if something went wrong.

Should I go ahead and merge the 'improve-tests' branch now that I have fixed the test_casper to re-allow --name?
